pkg/test/context.go

```diff
@@ -85,3 +100,40 @@ func (ctx *TestCtx) CleanupNoT() {
 func (ctx *TestCtx) AddFinalizerFn(fn finalizerFn) {
 	ctx.CleanUpFns = append(ctx.CleanUpFns, fn)
 }
+
+// TODO: figure out how to properly retrieve object kind info from object and enable
```

Comment:
What's wrong with `obj.GetObjectKind()`?

Reply:
For some reason, on most object, the objectkind is blank for us. Not sure if why. Here is example output with the logging enabled:

```
Resource type &TypeMeta{Kind:,APIVersion:,} with namespace/name test-memcached/memcached-operator created
Resource type &TypeMeta{Kind:,APIVersion:,} with namespace/name test-memcached/memcached-operator created
Resource type &TypeMeta{Kind:,APIVersion:,} with namespace/name test-memcached/memcached-operator created
Resource type &TypeMeta{Kind:,APIVersion:,} with namespace/name test-memcached/memcached-operator created
Resource type &TypeMeta{Kind:,APIVersion:,} with namespace/name test-memcached/example-memcached created
```

Reply:
The reason that we would need objectkind is because we have 4 different types of objects using the same name (service account, role, rolebinding, operator), so the output wouldn't the too helpful without the objectkind/typemeta

Reply:
`For some reason, on most object, the objectkind is blank for us.` 
this makes sense because that usrs are not required to fill out the `TypeMeta` filed of the object they create and I believe `GetObjectKind()` simply retrieves that `TypeMeta. Kind ` field. 

I recall there is a way to infer the type of the runtime object because every interface has its actual type associate with it.

Reply:
@AlexNPavel yeah, that's strange. I am unsure what's going on.

Reply:
Hmm, just did a test and it seems that the controller runtime is modifying the object. It I do `GetObjectKind()` before the `client.Create()`, I get the correct `TypeMeta`. If I do it after the call to `client.Create()`, the `TypeMeta` is empty...